contentcuration/contentcuration/frontend/channelEdit/vuex/clipboard/actions.js

```diff
@@ -0,0 +1,240 @@
+
+export function loadChannels(context) {
+  return context.dispatch('contentNode/loadClipboardTree', null, { root: true }).then(treeNodes => {
+    return context.dispatch('loadTreeNodes', treeNodes);
+  });
+}
+
+export function loadTreeNodes(context, treeNodes) {
+  const clipboardTreeId = context.rootGetters['clipboardRootId'];
+  const rootNodes = treeNodes.filter(node => node.parent === clipboardTreeId);
+
+  if (!rootNodes.length) {
+    return [];
+  }
+
+  const root = true;
+
+  // Find the channels we need to load
+  const channelIds = uniq(treeNodes.map(node => node.channel_id).filter(Boolean));
+
+  return promiseChunk(channelIds, 50, ids => {
+    // Load up the source channels
+    return context.dispatch('channel/loadChannelList', { ids }, { root });
+  })
+    .then(channels => {
+      // We need the channel tree for each source, which we need for copying,
+      // although we could probably delay this.
+      //
+      return promiseChunk(channelIds, 5, ids => {
+        return Promise.all(
+          ids.map(id => context.dispatch('contentNode/loadChannelTree', id, { root }))
+        );
+      }).then(() => channels);
+    })
+    .then(channels => {
+      // Add the channel to the selected state, it acts like a node
+      channels.forEach(channel => {
+        if (!(channel.id in context.state.selected)) {
+          context.commit('UPDATE_SELECTION_STATE', {
+            id: channel.id,
+            selectionState: SelectionFlags.NONE,
+          });
+        }
+      });
+
+      // Be sure to put these in after the channels!
+      treeNodes.forEach(node => {
+        if (!(node.id in context.state.selected)) {
+          context.commit('UPDATE_SELECTION_STATE', {
+            id: node.id,
+            selectionState: SelectionFlags.NONE,
+          });
+        }
+      });
+
+      // Do not return dispatch, let this operate async
+      context.dispatch('loadChannelColors');
+      return treeNodes;
+    });
+}
+
+// Here are the swatches Vibrant gives, and the order we'll check them for colors we can use.
+const colorChoiceOrder = [
+  'Vibrant',
+  'LightVibrant',
+  'DarkVibrant',
+  'LightMuted',
+  'Muted',
+  'DarkMuted',
+];
+
+export function loadChannelColors(context) {
+  const channels = context.getters.channels;
+
+  // Reducing the channels, going one by one processing colors, and collection an array
+  // of all of them
+  return channels.reduce((promise, channel) => {
+    const src = channel.thumbnail_encoding['base64']
+      ? channel.thumbnail_encoding['base64']
+      : channel.thumbnail_url;
+
+    // If we already have the color, just no src, then we'll just skip
+    if (context.getters.getChannelColor(channel.id, null) !== null || !src) {
+      return promise;
+    }
+
+    const image = new Image();
+    image.src = src;
+
+    //
+    return promise.then(allColors => {
+      return Vibrant.from(image)
+        .getPalette()
+        .then(palette => {
+          let color = '#6c939b';
```

Comment:
Could you add this to the vuetify theme color palette or if that's not doable with this being an action, could you create a constant somewhere so it's easier to track down if we need to change it later?

Reply:
I can do either, but I don't really know what this color is specifically. I pulled it from the old studio code.

Reply:
Ah okay, this is probably the dominant color that gets picked up from the Kolibri logo. Might be good to check with the design team what the default color should be